mp/src/game/server/client.cpp

```diff
@@ -847,52 +848,65 @@ CON_COMMAND( say_team, "Display player message to team" )
 	}
 }
 
-
-//------------------------------------------------------------------------------
-//------------------------------------------------------------------------------
-CON_COMMAND( give, "Give item to player.\n\tArguments: <item_name>" )
+static int WeaponCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
 {
-	CBasePlayer *pPlayer = ToBasePlayer( UTIL_GetCommandClient() ); 
-	if ( pPlayer 
-		&& (gpGlobals->maxClients == 1 || sv_cheats->GetBool()) 
-		&& args.ArgC() >= 2
-		&& !pPlayer->IsObserver() )
+	const auto pCmdName = "give_weapon";
+	char *substring = nullptr;
+	if (Q_strstr(pPartial, pCmdName) && strlen(pPartial) > strlen(pCmdName) + 1)
 	{
-		char item_to_give[ 256 ];
-		Q_strncpy( item_to_give, args[1], sizeof( item_to_give ) );
-		Q_strlower( item_to_give );
+		substring = (char *)pPartial + strlen(pCmdName) + 1;
+	}
 
-		// Don't allow regular users to create point_servercommand entities for the same reason as blocking ent_fire
-		if ( !Q_stricmp( item_to_give, "point_servercommand" ) )
+	int current = 0;
+	// Skip over weapon_none
+	for (int weaponID = WEAPON_PISTOL; weaponID < WEAPON_MAX; weaponID++)
```

Comment:
Bit of a nit, but can we add an alias enum value:
```cpp
enum WeaponID_t
{
  WEAPON_NONE = -1,

  WEAPON_FIRST,
  WEAPON_PISTOL = WEAPON_FIRST,
  // ...
}
```

Confusing ATM having to remember that `WEAPON_PISTOL` is the first entry.

Reply:
Resolved in https://github.com/momentum-mod/game/pull/624/commits/5028130d86e7e11fe9a8817e2f30261e0b227613